### ISSUE TITLE
gizmo init failed

### ISSUE THREAD

**Issue by hellolukeding** [Fennec-hub/three-viewport-gizmo#9]
thanks a lot for this lib,but i get some problem.First ,the dom has been rendered in document,but its content disappeared...

<img width="1512" alt="image" src="https://github.com/Fennec-hub/three-viewport-gizmo/assets/47938348/67ad6f29-6a55-4a90-a582-7814877e0830">
I think there is some mistakes,this is my code:

```typescript
  private controllerInit() {
     this.controls = new OrbitControls(this.camera!, this.container);
     this.viewportGizmo = new ViewportGizmo(this.camera!, this.renderer!,{
      placement:"bottom-right"
     })
     this.viewportGizmo.target=this.controls!.target;
    this.viewportGizmo.addEventListener("start", () => (this.controls!.enabled = false));
    this.viewportGizmo.addEventListener("end", () => (this.controls!.enabled = true));
    this.controls.addEventListener("change", () => {
       this.viewportGizmo!.update();
    });

    this.controls.enablePan = true;
    this.controls.enableZoom = true;
    // // controls.enableRotate = true;
    this.controls.enableDamping = true; // 作用是使动画很平滑
    this.controls.dampingFactor = 0.03;
    this.controls.zoomSpeed = 1;
    // // controls.rotateSpeed = 1;
    // //限制旋转角度
    this.controls.maxPolarAngle = Math.PI / 2;
    const tick = () => {
    this.viewportGizmo?.render();

      this.renderer!.render(this.scene!, this.camera!);
      if (this.controls!.enabled) this.controls!.update();
      requestAnimationFrame(tick);
    };
    tick();

  }

```
```typescript
import { Stats } from "stats.ts";
import * as THREE from "three";
import { ViewportGizmo } from "three-viewport-gizmo";
import { OrbitControls } from "three/examples/jsm/controls/OrbitControls";
class ViewerService {
  private container: HTMLDivElement;
  private scene?: THREE.Scene;
  private camera?: THREE.PerspectiveCamera;
  private renderer?: THREE.WebGLRenderer;
  private loadingManager: THREE.LoadingManager;
  private controls?: OrbitControls;
  private viewportGizmo?:ViewportGizmo;
  axesHelper?: THREE.AxesHelper;
  gridHelper?: THREE.GridHelper;

  constructor(container: HTMLDivElement | null) {
    if (!container) {
      throw new Error("container is null");
    }
    this.container = container;
    this.loadingManager = new THREE.LoadingManager();
  }

  /*--------------------------------------- 公共方法 ------------------------------------------*/
  init() {
    const renderer = new THREE.WebGLRenderer({
      antialias: true, //抗锯齿
      alpha: true, //透明背景
    });
    renderer.shadowMap.enabled = true;
    renderer.setSize(this.container.clientWidth, this.container.clientHeight);
    renderer.setPixelRatio(Math.min(window.devicePixelRatio, 2));
    this.container.appendChild(renderer.domElement);
    //
    const scene = new THREE.Scene();
    scene.background = new THREE.Color(0x252525);
    const camera = new THREE.PerspectiveCamera(
      75,
      this.container.clientWidth / this.container.clientHeight,
      0.1,
      99999999
    );
    camera.position.set(10000, 10000, 10000);
    camera.lookAt(0, 0, 0);

    this.scene = scene;
    this.camera = camera;
    this.renderer = renderer;

    this.helperInit();
    this.controllerInit();
    this.showStats();
  
    window.addEventListener("resize", () => {
      this.resize();
    });
  }

  getViewer() {
    if (!this.renderer || !this.scene || !this.camera) {
      throw new Error("viewer not init");
    }
    return {
      renderer: this.renderer,
      scene: this.scene,
      camera: this.camera,
    };
  }


  /*--------------------------------------- 私有方法 ------------------------------------------*/
  private helperInit() {
    const axesHelper = new THREE.AxesHelper(99999999);
    this.scene?.add(axesHelper);
    const gridHelper = new THREE.GridHelper(10000000, 10000);
    this.scene?.add(gridHelper);
    this.axesHelper = axesHelper;
    this.gridHelper = gridHelper;
  }

  private controllerInit() {
     this.controls = new OrbitControls(this.camera!, this.container);
     this.viewportGizmo = new ViewportGizmo(this.camera!, this.renderer!,{
      placement:"bottom-right"
     })
     this.viewportGizmo.target=this.controls!.target;
    this.viewportGizmo.addEventListener("start", () => (this.controls!.enabled = false));
    this.viewportGizmo.addEventListener("end", () => (this.controls!.enabled = true));
    this.controls.addEventListener("change", () => {
       this.viewportGizmo!.update();
    });

    this.controls.enablePan = true;
    this.controls.enableZoom = true;
    // // controls.enableRotate = true;
    this.controls.enableDamping = true; // 作用是使动画很平滑
    this.controls.dampingFactor = 0.03;
    this.controls.zoomSpeed = 1;
    // // controls.rotateSpeed = 1;
    // //限制旋转角度
    this.controls.maxPolarAngle = Math.PI / 2;
    const tick = () => {
    this.viewportGizmo?.render();

      this.renderer!.render(this.scene!, this.camera!);
      if (this.controls!.enabled) this.controls!.update();
      requestAnimationFrame(tick);
    };
    tick();

  }

  private showStats() {
    const stats = new Stats();
    stats.showPanel(0);
    this.container.appendChild(stats.dom);
    stats.dom.setAttribute(
      "style",
      "position: absolute; top: 0px;left: 0px;"
    );
    const tick = () => {
      stats.begin();
      stats.end();
      requestAnimationFrame(tick);
    };
    tick();
  }

  private resize() {

  }

}

export default ViewerService;
```

**Comment by dirkk0** [comment]
I had to put the gizmo render AFTER the camera render to make it visible.

**Comment by crystalfp** [comment]
Dear @dirkk0 , I have the call to `viewportGizmo.render();` after `renderer.render(scene, camera);` but nothing appears. also switching the two changes nothing. The viewport is here at top right and something does, if I drag the mouse inside.
If I change the opacity to 1, a pale disk appears but empty.
Do you have a working demo that I can study?
Thanks!

**Comment by dirkk0** [comment]
Hi @crystalfp ! The best I could come up with is here:
https://glitch.com/edit/#!/dk-threejs-gizmo
You can see that once you exchange the two render calls, the gizmo disappears.
Other than that I gave up on this lib, because there are several side effects that ruined it for me.

**Comment by Fennec-hub** [comment]
@hellolukeding, @crystalfp, @dirkk, 

I'm truly sorry for delayed response 🙏, I've been sick for the past months and slowly recovering... I've taken note of all the up mentioned issues,

- The relative positioning
- Orientation issue when the camera is not centered
- Clean up and better TS support
- Add maintainers

I'm working on it right now, the fixes should be ready for the next few days.

**Comment by Fennec-hub** [comment]
Fix [#13](https://github.com/Fennec-hub/three-viewport-gizmo/issues/13)
